tests/test_subprocess.py

```diff
@@ -56,6 +56,9 @@ def test_container(self):
         r = requests.get(self.url)
         self.assertIn('nginx', r.content)
 
+        requests.get(self.url + 'bad-path')
+        # TODO: What should happen here?
```

Comment:
Maybe just a `mock.patch.object(logger, "warn")`, and checking if that was called would suffice?